cirq/contrib/qasm_import/_parser.py

```diff
@@ -35,6 +36,41 @@ def __init__(self, supported_format: bool, qelib1_include: bool,
         self.circuit = c
 
 
+class QasmGate:
+
+    def __init__(self, qasm_gate: str, cirq_gate: cirq.Gate, num_args: int):
+        self.qasm_gate = qasm_gate
+        self.cirq_gate = cirq_gate
+        self.num_args = num_args
+
+    def validate_args(self, args, lineno):
+        if len(args) != self.num_args:
+            raise QasmException(
+                "{} only takes {} arg(s) (qubits and/or registers), "
+                "got: {}, at line {}".format(self.qasm_gate, self.num_args,
+                                             len(args), lineno))
+
+    def on(self, args: List[List[cirq.Qid]],
+           lineno: int = 0) -> Iterable[cirq.Operation]:
+        self.validate_args(args, lineno)
+        reg_size = 1
+        for reg in args:
+            if len(reg) > 1 and len(reg) != reg_size:
+                if reg_size == 1:
+                    reg_size = len(reg)
+                else:
+                    raise QasmException(
+                        "Non matching quantum registers of length {} "
+                        "at line {}".format([len(reg) for reg in args], lineno))
+
+        for i in range(reg_size):
+            # because of mixed quantum register/qubit support there can
+            # be single qubit registers and longer qubit registers
+            # for single qubit registers qreg[0], for longer ones qreg[i]
+            qubits = [qreg[min(len(qreg) - 1, i)] for qreg in args]
+            yield self.cirq_gate.on(*qubits)
+
+
 class QasmParser(object):
```

Comment:
No need to inherit from `Object`: https://stackoverflow.com/a/1238632.

Reply:
👍